openlibrary/plugins/openlibrary/home.py

```diff
@@ -51,22 +51,32 @@ def get_homepage():
     )
     return dict(page)
 
-def get_cached_homepage():
+
+def get_cached_homepage(lang='en'):
     five_minutes = 5 * dateutil.MINUTE_SECS
-    lang = web.ctx.get("lang", "en")
     return cache.memcache_memoize(
-        get_homepage, "home.homepage." + lang, timeout=five_minutes)()
+        get_homepage, 'home.homepage.' + lang, timeout=five_minutes
+    )()
+
 
 class home(delegate.page):
     path = "/"
 
     def GET(self):
-        cached_homepage = get_cached_homepage()
+        # Get the locale from the cookie or -if not found- set it to the default ('en')
+        locale_cookie = web.cookies().get('i18n_code')
+        if locale_cookie is not None:
```

Comment:
```suggestion
        if locale_cookie:
```
I think the `is not None` is redundant, but @cclauss would know for sure.

Reply:
For readability reasons  prefer the explicit check to `is not None`, but I have no strong opinion on this. Let me know what works best for you.

Reply:
Unfortunately, we still need support legacy Python so we can not use Python type hints.  How would we want the code to behave if `locale_cookie` was any other _falsey_ value like [], {}, set(), tuple(), 0, 0.0, False, etc.?  My sense is that all _falsey_ values are just as dangerous as None.

Reply:
Correct- all _falsey_ values should fall under that `if` clause. However, according to the official web.py (documentation)[], `web.cookies(https://webpy.org/cookbook/cookies).get(cookieName)` should return only `None`. Thus, the explicit check against that value.

Reply:
Hi, @cclauss. Is it OK if I leave this as is or should I go ahead and remove the `is not None` check?

Reply:
As this change is not a show-stopper and in an effort to merge the updated PR as soon as possible, I'm leaving this as is. Please don't hesitate to comment on the new PR, should you think that a change here is mandatory.